### DISCUSSION TITLE
How to test a private function in a C# class with xUnit and Visual stuidio

### DISCUSSION THREAD

**Discussion by TruePluto** [BYJRK/DotNet-Discussions#268] [C# 技术交流]
How to test a private function in a C# class with xUnit and Visual studio

**Comment by BYJRK** [comment]
一般有这么几种方式：

1. 使用 [InternalVisibleTo](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-8.0&redirectedfrom=MSDN) 特性
2. 使用反射
3. 考虑使用预编译器指令，使这些私有方法在特定情况下变为公有方法
4. 考虑将其封装到公有方法中，进而进行测试；或者这么理解，通常私有方法都是辅助公有方法的实现，所以通过测试这个公有方法，可以一定程度上测试所使用到的私有方法
